src/main/java/com/prykhodko/shop/utils/TotalPriceCounter.java

```diff
@@ -0,0 +1,29 @@
+package com.prykhodko.shop.utils;
+
+import com.prykhodko.shop.model.Basket;
+import com.prykhodko.shop.model.Product;
+import com.prykhodko.shop.service.ProductService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+
+@Service
+public class TotalPriceCounter {
+
+    private final ProductService productService;
+
+    @Autowired
+    public TotalPriceCounter(ProductService productService) {
+        this.productService = productService;
+    }
+
+    public BigDecimal count(Basket basket) {
```

Comment:
better move this method in basketDaoImpl class

Reply:
fixed